src/meili/map_matcher.cc

```diff
@@ -216,8 +217,9 @@ FindMatchResult(const MapMatching& mapmatching,
     const auto& next_state = mapmatching.state(next_stateid);
     // It must stay on the first edge of the route
     for (auto label = state.RouteBegin(next_state); label != state.RouteEnd(); label++) {
-      if (label->edgeid.Is_Valid()) {
-        edgeid = label->edgeid;
+      if (label->edgeid().Is_Valid()) {
```

Comment:
Thanks. I am not sure why the map matching logic requires the first label to have an invalid edge id (could it check if predecessor is invalid?). If that dependency can be removed then we can just use edgeid in the base class.